Classes/EventListener/Linkvalidator/BeforeRecordIsAnalyzedEventListener.php

```diff
@@ -0,0 +1,126 @@
+<?php
+
+declare(strict_types=1);
+
+namespace T3docs\Examples\EventListener\Linkvalidator;
+
+use TYPO3\CMS\Core\DataHandling\SoftReference\SoftReferenceParserFactory;
+use TYPO3\CMS\Linkvalidator\Event\BeforeRecordIsAnalyzedEvent;
+use TYPO3\CMS\Linkvalidator\Repository\BrokenLinkRepository;
+
+final class BeforeRecordIsAnalyzedEventListener
+{
+    protected BrokenLinkRepository $brokenLinkRepository;
+    protected SoftReferenceParserFactory $softReferenceParserFactory;
+    public function __construct(
+        BrokenLinkRepository $brokenLinkRepository,
+        SoftReferenceParserFactory $softReferenceParserFactory
+    ) {
+        $this->brokenLinkRepository = $brokenLinkRepository;
+        $this->softReferenceParserFactory = $softReferenceParserFactory;
+    }
+
+    private function addItemToBrokenLinkRepository(string $table, string $field, array $newsItem, string $foundUrl, string $forbiddenDomain): void
+    {
+        $link = [
+            'record_uid' => $newsItem['uid'],
+            'record_pid' => $newsItem['pid'],
+            'language' => $newsItem['sys_language_uid'],
+            'field' => $field,
+            'table_name' => $table,
+            'url' => $foundUrl,
+            'last_check' => time(),
+            'link_type' => 'external',
+        ];
+        $this->brokenLinkRepository->addBrokenLink($link, false, [
+            'errorType' => 'exception',
+            'exception' => 'Do not link externally to ' . $forbiddenDomain,
+            'errno' => 42, ]);
```

Comment:
Regarding "errno": It is not displayed in the module. It is stored in plain JSON in the table. If recommend to removed it, as it is not necessary. If this should be a "faked" HTTP status code, I recommend to use one of the 4xx range to be RFC complient. But I don't see the necessity. I also looked through other errors and not every error has an errno attached.

Reply:
A unique error number makes it easier to find where in the code the error was produced. Exchanged it with a current unix timestamp. We can mention that in a tip box in the docs